components/utils/DownMenu.tsx

```
@@ -0,0 +1,68 @@
+    { id: 3, name: '分类' },
+    { id: 4, name: '关于我们' },
+];
+
+export function DownMenu() {
+    return (
+        <div className="flex gap-3 transition-transform duration-1000">
+            {Menudata.map((item: any) => {
+                // ❗ 每个菜单单独创建自己的 ref
+                const triggerRef = useRef<HTMLButtonElement | null>(null);
+
+                return (
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)